liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -895,5 +923,29 @@ private boolean warXmlExists(File dir){
         }
         return looseApp;
     }
+    
+    private void listFiles(File directory, List<File> files, String suffix) {
+        if (directory != null) {
+            // Get all files from a directory.
+            File[] fList = directory.listFiles();
+            if (fList != null) {
+                for (File file : fList) {
+                    if (suffix != null) {
```

Comment:
You could replace lines 933 to 945 as follows to make use of short-circuit evaluation:
```
                        if (file.isFile() && ((suffix == null) || (file.getName().toLowerCase().endsWith("." + suffix)))) {
                            files.add(file);
                        } else if (file.isDirectory()) {
                            listFiles(file, files, suffix);
                        }
```